### PR TITLE
feat: add disabled state to reset configuration button and add successful reset toast

### PR DESCRIPTION
**Pull request by RyukTheCoder**
# Summary

Added disabled state to reset configuration button and added successful reset toast. Added "toast" variant to "Alert" component.

Fixes #1241

# How did you test this change?

Tested this by running playground on Chrome browser and checking reset configuration functionality.

# Checklist:

- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Implemented a user interface (UI) change, referencing our Figma design to ensure pixel-perfect precision.

### CODE REVIEW

**Review by yeager-eren — CHANGES_REQUESTED** [review]
I quickly went through you PR and I have 2 high level comments then I will do the actual code reveiw:

1. If it has enter animation, it also should have exit animation (toast).
2. Let's keep base component small as possible. You've added a `variant` to `Alert` which is actually not. Toast is a place that you are going to use Alert. Alert doesn't need to know where you are going to use. you may need to add more customization to Alert like `title` size which you can think and add it in a clean way, but let's move out your `toast variant` outside of Alert.

**Comment by RyukTheCoder** [comment]
> I quickly went through you PR and I have 2 high level comments then I will do the actual code reveiw:
> 
> 1. If it has enter animation, it also should have exit animation (toast).
> 2. Let's keep base component small as possible. You've added a `variant` to `Alert` which is actually not. Toast is a place that you are going to use Alert. Alert doesn't need to know where you are going to use. you may need to add more customization to Alert like `title` size which you can think and add it in a clean way, but let's move out your `toast variant` outside of Alert.

Done

File: widget/ui/src/components/Alert/Alert.tsx [hunk 1]
```diff
@@ -18,31 +23,29 @@ export function Alert(props: PropsWithChildren<PropTypes>) {
...
   } = props;
   const isFooterString = typeof footer === 'string';
 
   return (
-    <Container
-      className="_alert"
-      style={containerStyles}
-      type={type}
-      variant={variant}>
+    <Container css={containerStyles} type={type} variant={variant}>
```
Comment (yeager-eren) [inline widget/ui/src/components/Alert/Alert.tsx:33]: Why did you remove `className`?

File: widget/ui/src/components/Alert/Alert.types.ts [hunk 2]
```diff
@@ -8,5 +9,8 @@ export interface PropTypes {
   footer?: ReactNode;
   action?: ReactNode;
   variant?: 'alarm' | 'regular';
-  containerStyles?: CSSProperties;
+  containerStyles?: { [key: string]: string };
```
Comment (yeager-eren) [inline widget/ui/src/components/Alert/Alert.types.ts:12]: I think `css` from stiches has its own types. Could you please check?

File: widget/ui/src/components/Alert/Alert.types.ts [hunk 3]
```diff
@@ -8,5 +9,8 @@ export interface PropTypes {
...
+  titleContainerStyles?: { [key: string]: string };
```
Comment (yeager-eren) [inline widget/ui/src/components/Alert/Alert.types.ts:13]: Please remove this. this only using to handle `gap` style. you can skip this, it's a design mistake. we shouldn't handle.

File: widget/ui/src/components/Alert/Alert.types.ts [hunk 4]
```diff
@@ -8,5 +9,8 @@ export interface PropTypes {
...
+  titleSize?: BaseSizes;
```
Comment (yeager-eren) [inline widget/ui/src/components/Alert/Alert.types.ts:14]: I think that would better to accept a component as `title`, instead of adding Typography props to Alert.
Something like this:
```
<Alert 
title={<Typograph ... />}
/>
```

File: widget/playground/src/containers/configContainer/ConfigContainer.Header.tsx [hunk 5]
```diff
@@ -1,44 +1,80 @@
...
+        <ExportConfigModal
+          open={openExportModal}
+          onClose={toggleModal}
+          config={config}
+        />
+      </HeaderContainer>
+      <SuccessfulResetAlertContainer visible={isReset}>
+        <Alert
+          type="success"
+          title="The system's configuration data reset successfully"
```
Comment (yeager-eren) [inline widget/playground/src/containers/configContainer/ConfigContainer.Header.tsx:66]:
```suggestion
          title="Your applied configuration has been reset."
```

File: widget/playground/src/containers/configContainer/ConfigContainer.Header.tsx [hunk 6]
```diff
@@ -1,44 +1,80 @@
...
+import { isConfigChanged } from '../../utils/configs';
 
 import {
   HeaderContainer,
   ResetButton,
   StyledButton,
+  SuccessfulResetAlertContainer,
 } from './ConfigContainer.styles';
 
+const DISPLAY_SUCCESSFUL_RESET_TOAST_DURATION = 2_000;
```
Comment (yeager-eren) [inline widget/playground/src/containers/configContainer/ConfigContainer.Header.tsx:15]:
```suggestion
const TOAST_DURATION = 2_000;
```

File: widget/playground/src/containers/configContainer/ConfigContainer.Header.tsx [hunk 7]
```diff
@@ -1,44 +1,80 @@
...
+
 export function Header() {
   const resetConfig = useConfigStore.use.resetConfig();
   const [openExportModal, setOpenExportModal] = useState(false);
   const config = useConfigStore.use.config();
+  const [isReset, setIsReset] = useState(false);
```
Comment (yeager-eren) [inline widget/playground/src/containers/configContainer/ConfigContainer.Header.tsx:21]:
```suggestion
  const [isToastVisible, setIsToastVisible] = useState(false);
```

**Review by yeager-eren — CHANGES_REQUESTED** [review]
Thanks, We are getting there. Put some comments.

File: widget/ui/src/components/Alert/Alert.tsx (hunk 1 above)
Comment (RyukTheCoder) [reply widget/ui/src/components/Alert/Alert.tsx:33]: Sorry that was a mistake. fixed.

File: widget/ui/src/components/Alert/Alert.types.ts (hunk 2 above)
Comment (RyukTheCoder) [reply widget/ui/src/components/Alert/Alert.types.ts:12]: I fixed it. However, in some other files like "widget/embedded/src/components/TokenList/TokenList.types.ts" it is used this way. Maybe we should correct them too.

File: widget/ui/src/components/Alert/Alert.types.ts (hunk 3 above)
Comment (RyukTheCoder) [reply widget/ui/src/components/Alert/Alert.types.ts:13]: Removed

File: widget/ui/src/components/Alert/Alert.types.ts (hunk 4 above)
Comment (RyukTheCoder) [reply widget/ui/src/components/Alert/Alert.types.ts:14]: Done

File: widget/playground/src/containers/configContainer/ConfigContainer.Header.tsx (hunk 5 above)
Comment (RyukTheCoder) [reply widget/playground/src/containers/configContainer/ConfigContainer.Header.tsx:66]: I changed it. However the old title was what declared in design files.

File: widget/playground/src/containers/configContainer/ConfigContainer.Header.tsx (hunk 6 above)
Comment (RyukTheCoder) [reply widget/playground/src/containers/configContainer/ConfigContainer.Header.tsx:15]: Fixed

File: widget/playground/src/containers/configContainer/ConfigContainer.Header.tsx (hunk 7 above)
Comment (RyukTheCoder) [reply widget/playground/src/containers/configContainer/ConfigContainer.Header.tsx:21]: Fixed

File: widget/ui/src/components/Alert/Alert.tsx [hunk 8]
```diff
@@ -37,12 +42,11 @@ export function Alert(props: PropsWithChildren<PropTypes>) {
             <Typography
               color={getColor(type, variant)}
               variant="body"
-              className="title_typography"
-              size={variant === 'regular' ? 'xsmall' : 'small'}>
+              size={TitleSize[variant]}>
```
Comment (yeager-eren) [inline widget/ui/src/components/Alert/Alert.tsx:45]: You removed className here as well.

File: widget/ui/src/components/Alert/Alert.tsx [hunk 9]
```diff
@@ -37,12 +42,11 @@ export function Alert(props: PropsWithChildren<PropTypes>) {
             <Typography
               color={getColor(type, variant)}
               variant="body"
```
Comment (yeager-eren) [inline widget/ui/src/components/Alert/Alert.tsx:44]: You removed className here as well.

File: widget/ui/src/components/Alert/Alert.tsx [hunk 10]
```diff
@@ -37,12 +42,11 @@ export function Alert(props: PropsWithChildren<PropTypes>) {
...
-              className="title_typography"
```
Comment (yeager-eren) [inline widget/ui/src/components/Alert/Alert.tsx:40]: Don't use `enum` for this purpose. you can do something like this:

```
function mapVariantToSize(variant){
switch(variant){
...
}
}
```

File: widget/ui/src/theme.ts [hunk 11]
```diff
@@ -1,6 +1,6 @@
 // TSDX + stiches doesn't work with import statement and has a bug.
 
-import type { PropertyValue } from '@stitches/react';
+import type * as Stitches from '@stitches/react';
```
Comment (yeager-eren) [inline widget/ui/src/theme.ts:3]: For better tree shaking, I suggest something like this:
```
import type { PropertyValue, CSS as StitchesCSS } from '@stitches/react';
```

**Review by yeager-eren — CHANGES_REQUESTED** [review]
Thanks for doing changes. I put some minor comments.

File: widget/ui/src/components/Alert/Alert.tsx (hunk 8 above)
Comment (RyukTheCoder) [reply widget/ui/src/components/Alert/Alert.tsx:45]: fixed

File: widget/ui/src/theme.ts (hunk 11 above)
Comment (RyukTheCoder) [reply widget/ui/src/theme.ts:3]: done

File: widget/ui/src/components/Alert/Alert.tsx (hunk 9 above)
Comment (RyukTheCoder) [reply widget/ui/src/components/Alert/Alert.tsx:39]: fixed

File: widget/ui/src/components/Alert/Alert.helpers.ts [hunk 12]
```diff
@@ -15,3 +15,12 @@ export const getColor = (type: Type, variant: 'alarm' | 'regular') => {
       return undefined;
   }
 };
+
+export const mapVariantToSize = (variant: 'alarm' | 'regular') => {
```
Comment (yeager-eren) [inline widget/ui/src/components/Alert/Alert.helpers.ts:19]: Please use `PropTypes["variant"]` for the type.

**Review by yeager-eren — APPROVED** [review]
Thanks.

File: widget/ui/src/components/Alert/Alert.helpers.ts (hunk 12 above)
Comment (RyukTheCoder) [reply widget/ui/src/components/Alert/Alert.helpers.ts]: Done

**Comment by samobasquiat** [comment]
in `widget/playground/src/App.tsx:
`
```
<ToastProvider container={document.body}>
    <div id={PLAYGROUND_CONTAINER_ID} className={activeStyle}>
      <WidgetProvider config={overridedConfig}>
        <ConfigContainer>
          <Routes>
            <Route path="/*" element={<Widget config={overridedConfig} />} />
          </Routes>
        </ConfigContainer>
      </WidgetProvider>
    </div>
    </ToastProvider>
```


`  const { addToast } = useToast();
`

```
onClick={() => {
            setIsToastVisible(true);
            resetConfig();
            addToast({
              title: 'HIIIIIIIII',
              type: 'success',
              position: 'center-bottom',
              containerStyle: {
                bottom: '10px'
              },
            })
          }}
```

**Comment by RyukTheCoder** [comment]
> in `widget/playground/src/App.tsx: ` you can wrap the whole app:
> 
> ```
> <ToastProvider container={document.body}>
>     <div id={PLAYGROUND_CONTAINER_ID} className={activeStyle}>
>       <WidgetProvider config={overridedConfig}>
>         <ConfigContainer>
>           <Routes>
>             <Route path="/*" element={<Widget config={overridedConfig} />} />
>           </Routes>
>         </ConfigContainer>
>       </WidgetProvider>
>     </div>
>     </ToastProvider>
> ```
> 
> And whenever you want to toast, can import addToast and use it. It gets several args and can check it. ` const { addToast } = useToast();`
> 
> ```
> onClick={() => {
>             resetConfig();
>             addToast({
>               title: 'HIIIIIIIII',
> autoHideDuration: 3000,
>               type: 'success',
>               position: 'center-bottom',
>               containerStyle: {
>                 bottom: '10px'
>               },
>             })
>           }}
> ```

I changed my code to use Toast component as you said. I just didn't revert my changes to Alert component because those are just minor refactoring changes. Let me know if I should revert them too.

**Review by samobasquiat — APPROVED** [review]
